### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by lazukr**
Updated readme to include JSON format for fireworks display sequence.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
 
 # Fireworks Display Sequence Format
 
-Format will be in the following form
+Fireworks display sequence format will be in JSON. The general format is as follows:
 ```
-[Timestamp] TypeOfFirework1 xPosition Angle Speed, TypeOfFirework2 xPosition Angle Speed, etc...
-[Timestamp] ...
-etc...
+{display: [
```
Comment (jsatgit) [inline README.md]: i like this formatting:

```
{
    "key": "value"
}
```

more than
```
{"key": "value"
}
```

File: README.md [hunk 2]
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
...
+  {timestamp: 2342342342, shells: [{type: 1, xPosition: 234, angle: 90, speed: 120}]},
```
Comment (jsatgit) [inline README.md]: The keys are strings so put double quotes around them like `"timestamp"`

File: README.md [hunk 3]
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
...
+  {timestamp: 12312123, shells: [{type: 1, xPosition: 232, angle: 45, speed: 50}, {type: 2, xPosition: 342, angle: 60, speed: 76}]}
```
Comment (jsatgit) [inline README.md]: Please format large array items as follows:

```
[
   {"type": 1, "xPosition": 232},
   {"type": 2, "xPosition": 242}
]
```

File: README.md (hunk 1 above)
Comment (jsatgit) [inline README.md]: Why do we have a single top level key called `display`? Are there any other top level keys? If we don't have other top level keys then this object should just be an array to keep it simple.

File: README.md (hunk 3 above)
Comment (jsatgit) [inline README.md]: I think the initial speed of a shell is a property of the shell. So shouldn't that be included in the type?

File: README.md [hunk 4]
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
...
 
-* [0000100] is a valid timestamp
-* 0005000 is not a valid timestamp
+This is an integer value that is measured in milliseconds.
 
-## TypeOfFirework
+## Type
 
-This is the specific firework that is to be fired. A database of all existing fireworks will be made, each with a unique identifier. TypeOfFirework will give us details regarding the number of stars used, the colour, type of stars used etc. A TypeOfFirework can be made of TypeOfFirework (the stars are fireworks themselves).
+This is the type of firework that is to be fired. A generic set of fireworks will be available. Custome fireworks can also be made. The fireworks details will also be in JSON format and requires a unique identifier. The type gives us information regarding the number of stars used, the type of stars used, the colour etc.
```
Comment (jsatgit) [inline README.md]: When you say custom ones can also be made, do you mean we are going to build a firework GUI for people to customize their fireworks? What about the explosion delay? Is that going to be included in the type of firework as well?

Comment (jsatgit) [inline README.md]: We discussed before that frequencies will be mapped to the `xPosition` and amplitude to height of explosion and power of explosion. Here we have no notion of height but we have `explosionDelay`. Can we incorporate that?

**Review by jsatgit — CHANGES_REQUESTED** [review]

**Comment by jsatgit** [comment]
In general please give your branch a more descriptive name than `patch-1`

File: README.md (hunk 1 above)
Comment (lazukr) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (lazukr) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (lazukr) [reply README.md]: display will be kept as the definition of various shells will be included in the fireworks display sequence.

File: README.md (hunk 4 above)
Comment (lazukr) [reply README.md]: user defined fireworks will be done using JSON as well. Fireworks themselves are in JSON.

File: README.md (hunk 3 above)
Comment (lazukr) [reply README.md]: Done.

Comment (lazukr) [reply README.md]: Initial speed will be part of the shell property.

File: README.md (hunk 4 above)
Comment (lazukr) [reply README.md]: burstDelay has been included as a property of the shell.

Comment (jsatgit) [reply README.md]: the "actual user" should not have to manipulate the JSON directly. What I'm asking is that, are we planning to build a GUI later on that allows the user to add custom fireworks.

File: README.md [hunk 5]
```diff
@@ -58,24 +58,56 @@ This will be the algorithm that will automatically generate a fireworks display
...
 
-Format will be in the following form
+Fireworks display sequence format will be in JSON. The general format is as follows:
 ```
-[Timestamp] TypeOfFirework1 xPosition Angle Speed, TypeOfFirework2 xPosition Angle Speed, etc...
-[Timestamp] ...
-etc...
+{
+  "shellDefinitions": [
+    {"type":1, "specs": {
```
Comment (jsatgit) [inline README.md]: one key per line please. Also the first key should go on the next line

File: README.md [hunk 6]
```diff
@@ -58,24 +58,56 @@ This will be the algorithm that will automatically generate a fireworks display
...
+                          "burstPower": 40,
+                          "burstDelay": 500
+                        }
+    ],
+  "display": [
+    {
+      "timestamp": 2342342342, 
+      "shells": [
+                  {"type": 1, "xPosition": 234, "angle": 90}
+                ]
```
Comment (jsatgit) [inline README.md]: i would prefer to format it like:
```
  "shells": [
    {"type": 1, "xPosition": 234, "angle": 90}
  ]

```

**Comment by jsatgit** [comment]
Please wrap json inside of triple back ticks so markdown formats it as code

**Comment by jsatgit** [comment]
check this out @lazukr : http://jsonformatter.org/

File: README.md [hunk 7]
```diff
@@ -59,39 +59,44 @@ This will be the algorithm that will automatically generate a fireworks display
...
+    {
+      "type":1, 
+      "specs": {
+        "numOfStars": 10,
+        "typeOfStars": "circular",
+        "colour": "blue",
+        "size": 10,
+        "burstPower": 20,
+        "burstDelay": 300
+      }
```
Comment (jsatgit) [inline README.md]: there needs to be a comma here

Comment (jsatgit) [reply README.md]: also ur indent is off

File: README.md [hunk 8]
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
...
+```JSON
+{
+  "shellDefinitions": [
+    {
+      "type":1, 
+      "specs": {
+        "numOfStars": 10,
+        "typeOfStars": "circular",
+        "colour": "blue",
+        "size": 10,
```
Comment (jsatgit) [inline README.md]: is this the radius of the stars or the radius of the burst?

File: README.md [hunk 9]
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
...
+      "specs": {
+        "numOfStars": 50,
+        "typeOfStars": "circular",
+        "colour": "red",
+        "size": 15,
+        "burstPower": 40,
+        "burstDelay": 500
+    }
+  ],
+  "display": [
```
Comment (jsatgit) [inline README.md]: let's use `sequence` instead of `display` ?

Comment (lazukr) [reply README.md]: aye boss.

File: README.md (hunk 8 above)
Comment (lazukr) [reply README.md]: this is size of shell.

File: README.md [hunk 10]
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
...
 # Fireworks Display Sequence Format
 
-Format will be in the following form
+Fireworks display sequence format will be in JSON. The general format is as follows:
+
+```JSON
+{
+  "shellDefinitions": [
+    {
+      "type":1,
```
Comment (jsatgit) [inline README.md]: there should be a space between the colon and the 1

**Review by jsatgit — APPROVED** [review]

**Pull request merged**